microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/core/java/JavaFeaturesRegistry.java

```diff
@@ -154,6 +158,16 @@ public List<JavaWorkspaceSymbolsDefinition> getJavaWorkspaceSymbolsDefinitions()
 		return javaWorkspaceSymbolsDefinitions;
 	}
 
+	/**
+	 * Returns a list of JAX-RS info definitions.
+	 *
+	 * @return a list of JAX-RS info definitions
+	 */
+	public List<JaxRsInfoDefinition> getJaxRsInfoDefinitions() {
+		loadJavaFeatureDefinitions();
+		return jaxRsInfoDefinitions;

```

Comment:
I will define a boolean  defaultJaxRsInfoProviderAdded in the class to write:

```java
if (!defaultJaxRsInfoProviderAdded) {
  jaxRsInfoProviders.add(new DefaultJaxRsInfoProvider());
  defaultJaxRsInfoProviderAdded = true;
}
```

Reply:
If I always add the default provider right after the extensions are loaded, does it still make sense to do this?

Reply:
You need to add DefaultJaxRsInfoProvider just one time at the endof the list, no?

Reply:
Oh right, I see what you mean. I have seen your code, it is perfect.